tests/test_app_integration.py

```diff
@@ -173,3 +182,74 @@ def test_delete_post_endpoint(mock_mongoDb_client, client):
     assert response.status_code == 302
     assert "<h1>Redirecting...</h1>" in response_html
 
+@mock.patch('pymongo.MongoClient', side_effect=mock_mongoDb_client)
+def test_admin_endpoint(mock_mongoDb_client, client):
+    #given
+
+    # when
+    response = client.get("/admin")
+    response_html = response.data.decode()
+
+    # then 
+    assert response.status_code == 200
+    assert "<title>To-Do App</title>" in response_html
+    assert "<h2>Users</h2>" in response_html
+    assert "<td>50665</td>" in response_html
+    assert "<td>pcoulso1</td>" in response_html
+    assert "<td>50666</td>" in response_html
+    assert "<td>pcoulso2</td>" in response_html
+    assert "<td>Peter</td>" in response_html
+    assert "<td>50667</td>" in response_html
+    assert "<td>pcoulso3</td>" in response_html
+    assert "<td>Paul</td>" in response_html
+    assert "<td>50668</td>" in response_html
+    assert "<td>pcoulso4</td>" in response_html
+    assert "<td>Pat</td>" in response_html
+
+@mock.patch('pymongo.MongoClient', side_effect=mock_mongoDb_client)
+def test_user_delete_post_endpoint(mock_mongoDb_client, client):
+    #given
+
+    # when
+    response = client.post("/user/delete/50665")
+    response_html = response.data.decode()
+
+    # then 
+    assert response.status_code == 302
```

Comment:
It's worth noting that this test is testing that we're redirected when we try delete a user - it doesn't actually have any sense of whether that action succeeded. This could cause confusion!

Reply:
I'm not really sure what can be done to test the outcome here. Do you think I should remove the tests ?

Reply:
Happy to leave it as is for now, but it would be better if we then checked its status, either by visiting the index page & checking the status there, or probably more easily by verifying that the item has changed in the mock mongo database